### ISSUE TITLE
[3.3.5] Improved Corruption calculation error

### ISSUE THREAD

**Issue by Keader** [TrinityCore/TrinityCore#14812]
1 - Improved Corruption

Spell Rankings:
http://wotlk.openwow.com/spell=17810
http://wotlk.openwow.com/spell=17811
http://wotlk.openwow.com/spell=17812
http://wotlk.openwow.com/spell=17813
http://wotlk.openwow.com/spell=17814

Description:  "Increases the damage done by your Corruption by x%, and increases the critical strike chance of your Seed of Corruption by y%."

Problem: The calculation in damage offered by the spell talent in corruption, is below promised by the ranking.

My test

Character Status: 
![wowscrnshot_053015_201930](https://cloud.githubusercontent.com/assets/7120913/7899817/ec456a80-0709-11e5-8412-6f7112404b65.jpg)

No points in this talent is the base damage:
![st](https://cloud.githubusercontent.com/assets/7120913/7899823/7968d960-070a-11e5-8a42-53dc55ef8edc.jpg)

1 Talent Point:  2% buff
![1p](https://cloud.githubusercontent.com/assets/7120913/7899832/c219c6c4-070a-11e5-9559-5229fe96f199.jpg)

Note: Note that the buff says the corruption of the damage is expected to increase 2%. However the test he increased approximately 1.6%.
value in the test : 1007 + 1.6% = 1023.11 
value that should have been obtained: 1007 + 2% = 1027.14

2 Talent Point: 4% buff
![2p](https://cloud.githubusercontent.com/assets/7120913/7900292/ac06c850-0723-11e5-927e-0c59993bb83d.jpg)

value in the test : 1007 + 3.2% = 1039,22
value that should have been obtained: 1007 + 4% = 1047,28

3 Talent Point: 6% buff
![p3](https://cloud.githubusercontent.com/assets/7120913/7900313/f8591e14-0724-11e5-8e79-5214f514124d.jpg)

value in the test : 1007 + 4.8% = 1055,33
value that should have been obtained: 1007 + 6% = 1067,42

4 Talent Point: 8% buff
![4p](https://cloud.githubusercontent.com/assets/7120913/7900322/728ffc52-0725-11e5-869e-c1baa1041881.jpg)

value in the test : 1007 + 6.4% = 1071,44
value that should have been obtained: 1007 + 8% = 1087,56

5 Talent Point: 10% buff
![5p](https://cloud.githubusercontent.com/assets/7120913/7900330/18b91a6e-0726-11e5-9e11-50315599997c.jpg)

value in the test : 1007 + 8% = 1087,56
value that should have been obtained: 1007 + 10% = 1107,7

The tests were performed on Heroic Dummy Ironforge (the military ward) with curse of elements in the Dummy.
I did not use any proc item not change results

TrinityCore rev. 5bd7c295c0b0 2015-05-13 22:29:49 +0200 (3.3.5 branch) (Win64, Release)

**Comment by jackpoz** [comment]
can you try without Cure of the elements and on the lowest level dummy ?

**Comment by Keader** [comment]
Oks

No curse of elements and removed the aura 46376 (Flask of the Frost Wyrm) too
Tested on Expert's Training Dummy (level 60) - Entry: 32666

No Talent (Base damage):

![st base](https://cloud.githubusercontent.com/assets/7120913/7901836/5e82cbce-0773-11e5-90a4-e6e7001646a2.jpg)

1 Talent Point: 2%

![1 pt](https://cloud.githubusercontent.com/assets/7120913/7901844/c71c2518-0773-11e5-8ad8-3d491994cfe5.jpg)


value in the test : 866 + 1.6% = 879,85
value that should have been obtained: 866 + 2% = 883,32

2 Talent Point: 4%

![2 pt](https://cloud.githubusercontent.com/assets/7120913/7901853/0938139e-0774-11e5-8aa7-aa6275fa3df8.jpg)

value in the test : 866 + 3,2% = 893,71
value that should have been obtained: 866 + 4% = 900,64

3 Talent Point: 6%

![3 pt](https://cloud.githubusercontent.com/assets/7120913/7901866/51b54268-0774-11e5-9e3f-a5d3be138b26.jpg)

value in the test : 866 + 4,8% = 907,56
value that should have been obtained: 866 + 6% = 917,96

4 Talent Point: 8%

![4 pt](https://cloud.githubusercontent.com/assets/7120913/7901867/5bc14af4-0774-11e5-947c-f7f4e2d97cc9.jpg)

value in the test : 866 +  6.4% = 921,42
value that should have been obtained: 866 + 8% = 935,28

5 Talent Point: 10%

![5 pt](https://cloud.githubusercontent.com/assets/7120913/7901872/a23786f6-0774-11e5-9be1-1befb5e4436d.jpg)

value in the test : 866 +  8% = 935,28
value that should have been obtained: 866 + 10% = 952,6

Recalling that some calculations are approximate values

**Comment by jackpoz** [comment]
Can you tried with no gear either ? I couldn't get any bonus at all with just the talent and no gear

**Comment by Keader** [comment]
Yes, I will remove all items. Post screenshots soon

**Comment by Keader** [comment]
wow ! Without gear level and without buffs, the talent just does not work. The base Damage (untalented) was 180. After placing ranking by ranking the damage remained at 180, as if the talent was not present

**Comment by robinsch** [comment]
SPELLMOD_DOT is applied in SpellDamageBonusDone / SpellHealingBonusDone but never called for !isAreaAura.

I've only tested it without gear, working fine.
```diff
diff --git a/src/server/game/Spells/Auras/SpellAuraEffects.cpp b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
index 34024d7..93c2193 100644
--- a/src/server/game/Spells/Auras/SpellAuraEffects.cpp
+++ b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
@@ -5843,8 +5843,13 @@ void AuraEffect::HandlePeriodicDamageAurasTick(Unit* target, Unit* caster) const
         if (isAreaAura)
             damage = caster->SpellDamageBonusDone(target, GetSpellInfo(), damage, DOT, GetBase()->GetStackAmount()) * caster->SpellDamagePctDone(target, m_spellInfo, DOT);
         else
+        {
             damage = std::max(int32(damage * GetDonePct()), 0);
 
+            if (Player* modOwner = caster->GetSpellModOwner())
+                modOwner->ApplySpellMod(GetSpellInfo()->Id, SPELLMOD_DOT, damage);
+        }
+
         damage = target->SpellDamageBonusTaken(caster, GetSpellInfo(), damage, DOT, GetBase()->GetStackAmount());
 
         // Calculate armor mitigation
@@ -6146,8 +6151,13 @@ void AuraEffect::HandlePeriodicHealAurasTick(Unit* target, Unit* caster) const
         if (isAreaAura)
             damage = caster->SpellHealingBonusDone(target, GetSpellInfo(), damage, DOT, GetBase()->GetStackAmount()) * caster->SpellHealingPctDone(target, m_spellInfo);
         else
+        {
             damage = std::max(int32(damage * GetDonePct()), 0);
 
+            if (Player* modOwner = caster->GetSpellModOwner())
+                modOwner->ApplySpellMod(GetSpellInfo()->Id, SPELLMOD_DOT, damage);
+        }
+
         damage = target->SpellHealingBonusTaken(caster, GetSpellInfo(), damage, DOT, GetBase()->GetStackAmount());
     }
```

**Comment by Keader** [comment]
@Aokromes fixed by #14817 
Fixed also the talents: Siphon Life, Shadow Mastery and Contagion. Who had similar problems.
